run_bin_test.go

```diff
@@ -396,6 +397,140 @@ func TestCoverageCollector_writeArgs(t *testing.T) {
 	}
 }
 
+func TestPreExec(t *testing.T) {
+	type args struct {
+		binPath      string
+		mainTestName string
+		env          []string
+		args         []string
+	}
+	var (
+		buffer1, buffer2 string
+	)
+
+	tests := []struct {
+		name        string
+		preCmdFuncs []PreCmdFunc
+		expected    string
+		args        args
+		wantErr     bool
+		wantErrMsg  string
+		buffer      *string
+	}{
+		{
+			name: "test running one prefunc",
```

Comment:
Do we need this test case?

Reply:
nope you're right, testing multiple prefuncs is sufficient